Library/Homebrew/utils/github.rb

```
@@ -371,6 +372,26 @@ def self.get_artifact_url(workflow_array)
     artifact.last["archive_download_url"]
   end
 
+  # Downloads an artifact from GitHub Actions.
+  #
+  # @param url [String] URL to download from
+  # @param artifact_id [String] a value that uniquely identifies the downloaded artifact
+  #
+  # @api private
+  sig { params(url: String, artifact_id: String).void }
+  def self.download_artifact(url, artifact_id)
+    odie "Credentials must be set to access the Artifacts API" if API.credentials_type == :none
```

Should probably raise a proper error here instead of just `odie`ing.